src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1226,21 +1226,26 @@ private ImmutableArray<MethodSymbol> ExplicitlyOverriddenMethods()
             return new MetadataDecoder(moduleSymbol, _containingType).GetExplicitlyOverriddenMethods(_containingType.Handle, _handle, this.ContainingType);
         }
 
-        internal MethodSymbol ExplicitlyOverriddenClassMethod
+        internal override MethodSymbol ExplicitlyOverriddenClassMethod
         {
             get
             {
                 if (IsExplicitClassOverride)
                 {
+                    // If there is a *single* methodimpl entry, we return that.
+                    MethodSymbol result = null;
                     foreach (MethodSymbol method in ExplicitlyOverriddenMethods())
                     {
                         if (method.ContainingType.IsClassType())
                         {
-                            return method;
+                            if (result is { })
+                                return null;
+
+                            result = method;
```

Comment:
>result = method [](start = 28, length = 15)

Do we care if the name matches?

Reply:
No.

---
In reply to: [424661333](https://github.com/dotnet/roslyn/pull/44025#discussion_r424661333) [](ancestors = 424661333)

Reply:
>> Do we care if the name matches?

> No.

It feels like there are interesting scenarios around that. Do we test any?

---
In reply to: [425494566](https://github.com/dotnet/roslyn/pull/44025#discussion_r425494566) [](ancestors = 425494566,424661333)

Reply:
Yes. See, for example, `ExplicitOverrideWithoutCSharpOverride` in `CodeGenOverridingAndHiding.cs`.

---
In reply to: [425496726](https://github.com/dotnet/roslyn/pull/44025#discussion_r425496726) [](ancestors = 425496726,425494566,424661333)